pyat/at/physics/radiation.py

```diff
@@ -22,6 +24,57 @@
                   ('emitXYZ', numpy.float64, (3,))]
 
 
+def _cumulb(it):
+    """accumulate diffusion matrices"""
+    cumul = numpy.zeros((6, 6))
+    yield cumul
+    for el, orbin, b in it:
+        m = find_elem_m66(el, orbin)
+        cumul = m.dot(cumul).dot(m.T) + b
+        yield cumul
+
+
+def _dmatr(ring, orbit=None, keep_lattice=False):
+    """
+    compute the cumulative diffusion and orbit
+    matrices over the ring
+    """
+    nelems = len(ring)
+    energy = ring.energy
+    allrefs = uint32_refpts(range(nelems + 1), nelems)
+
+    if orbit is None:
+        orbit, _ = find_orbit6(ring, keep_lattice=keep_lattice)
+        keep_lattice = True
+
+    orbs = numpy.squeeze(
+        lattice_pass(ring, orbit.copy(order='K'), refpts=allrefs,
+                     keep_lattice=keep_lattice), axis=(1, 3)).T
+    b0 = numpy.zeros((6, 6))
+    bb = [find_mpole_raddiff_matrix(elem, orbs[i], energy)
+          if elem.PassMethod.endswith('RadPass') else b0
+          for i, elem in enumerate(ring)]
```

Comment:
Why not using a single iterator for both `ring` and `orbs`? It's more efficient…
```suggestion
    bb = [find_mpole_raddiff_matrix(elem, elemorb, energy)
          if elem.PassMethod.endswith('RadPass') else b0
          for elem, elemorb in zip(ring,orbs)]
```

Reply:
Good point thanks!